plugin/dns64/dns64.go

```diff
@@ -0,0 +1,188 @@
+// Package dns64 implements a plugin that performs DNS64.
+package dns64
+
+import (
+	"context"
+	"errors"
+	"net"
+	"time"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/coredns/coredns/plugin/pkg/response"
+	"github.com/coredns/coredns/plugin/pkg/upstream"
+	"github.com/coredns/coredns/request"
+	"github.com/coredns/proxy"
+
+	"github.com/miekg/dns"
+)
+
+// DNS64 performs DNS64.
+type DNS64 struct {
+	*proxy.Proxy
+	NativeUpstream *upstream.Upstream
+	Next           plugin.Handler
+	Prefix         *net.IPNet
+	TranslateAll   bool // Not comply with 5.1.1
+}
+
+// ServeDNS implements the plugin.Handler interface.
+func (dns64 DNS64) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	hijackedWriter := &ResponseWriter{w, dns64, ctx, r}
```

Comment:
this needs to do more; like check the zones it's authoritative for and other bits; looks too short